ruby/examples/lib/apply_order_discount.rb

```diff
@@ -0,0 +1,44 @@
+class ApplyOrderDiscount
+  Request = Struct.new(:id)
+  Response = Struct.new(:discount, :has_discount?)
+
+  attr_reader :gateway
+
+  def initialize(gateway: Gateway.new)
+    @gateway = gateway
+  end
+
+  def apply(request)
+    order = gateway.find_order_by_id(request.id)
+
+    unless order.has_discount?
+      order.discount = discount_amount
+      gateway.save_order_discount(request.id, order)
+    end
+
+    Response.new(order.discount, order.has_discount?)
+  end
+
+  private
+
+  def discount_amount
+    10
+  end
+
+  class DiscountableOrder < Struct.new(:discount)
+    def has_discount?
+      discount > 0
+    end
+  end
+
+  class Gateway
+    def find_order_by_id(id)
+      order = Spree::Order.find(id)
+      DiscountableOrder.new(order.discount)
+    end
+
+    def save_order_discount(id, order)
+      Spree::Order.find(id).update!(discount: order.discount)
+    end
+  end
+end
```

Comment:
I think having it inside the same file helps show people that it's fairly easy to start with Clean Architecture, it shows the various levels of separation in one file, it makes the module as a whole easier to understand.

By taking this approach, we're showing rubyists it's a relatively low barrier to entry even for CRUD use cases.